### DISCUSSION TITLE
A couple issues to do with Song Importing

### DISCUSSION THREAD

**Discussion by JakeHRobinson** [ppy/osu#16133] [Q&A]
I've ran into a couple issues today. Basically, since I last tried to import my songs, I've downloaded about 60,000 maps. To account for the file size I had to symlink a folder to another drive so I'd be able to hold them. I've downloaded other songs through supporter since. When I go to import songs on osu!lazer, it says there's only 254 to import, clearly this isn't the case. So my first issue is I'm wondering since it's a symlink what do I need to do to be able to import them? 

The second issue is that for those 254 beatmaps, I get to 253/254 and I just get stuck importing. I've been importing that last song for an hour now, I've tried cancelling and restarting the import, it goes back to 253 in a few seconds and picks up where it left off.

**Comment by bdach** [comment]
254 is suspiciously close to 256 (say, `.`, `..` and then 254 other files). what is the filesystem on the drive to which your symlink points to?

**Reply by JakeHRobinson** [reply]
What do you mean by "what is the file system"?

**Reply by bdach** [reply]
https://en.wikipedia.org/wiki/File_system

examples: fat, ntfs (more windows-native), ext2, ext4 (more linux-native), apfs (apple devices)

**Reply by bdach** [reply]
also please attach logs as instructed in https://github.com/ppy/osu/discussions/15188

**Reply by JakeHRobinson** [reply]
The file system says "NTFS" when I check the properties of the drive. I should probably mention that I'm on windows 11 which I'm assuming it's listed as supported, however osu!lazer wasn't recognising the symlink when I was still on windows 10. Songs downloaded after I symlinked, handled through osu putting file in the right place or downloading ingame appear to be recognised if I go to lazer and import. 


[database.log](https://github.com/ppy/osu/files/7736694/database.log)
[network.log](https://github.com/ppy/osu/files/7736695/network.log)
[performance.log](https://github.com/ppy/osu/files/7736691/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7736692/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7736693/updater.log)

**Reply by bdach** [reply]
i wouldn't expect a ntfs-ntfs symlink to cause problems. so it may be dependent on the map itself that the import is getting stuck on. the logs unfortunately don't show which one it is.

**Reply by JakeHRobinson** [reply]
Well, the thing is there's definitely more than 254 maps to be imported, there's about 55,000 or so. 254 seems to be the number of maps I've downloaded since the symlink. This isn't the first time I've imported after the symlink, I've done it once before and there were about 150 maps then and it imported those fine. So it's recognising maps I've downloaded after the symlink, but not the ones from before. I downloaded about 60,000 maps at once in a big torrent, realised the file was too big, then symlinked my songs folder.
